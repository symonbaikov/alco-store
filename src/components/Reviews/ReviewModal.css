.review-modal {
    background: white;
    padding: 40px;
    width: 800px;
    max-width: 90%;
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    box-shadow: 0 4px 20px rgba(0, 0, 0, 0.15);
    max-height: 90vh;
    overflow-y: auto;
  }
  

  
  .review-modal-author {
    font-size: 28px;
    color: #333;
    margin: 0 0 12px 0;
    font-weight: 500;
  }
  
  .review-modal-rating {
    display: flex;
    gap: 4px;
    margin-bottom: 20px;
  }
  
  .review-modal-rating .star {
    font-size: 24px;
  }
  
  .review-modal-rating .star.filled {
    color: #ffd700;
  }
  
  .modal-close-button svg {
    width: 20px;
    height: 20px;
  }

  .review-modal-text {
    font-size: 17px;
    line-height: 1.8;
    color: #444;
    margin: 0;
    padding: 20px 0;
    white-space: pre-wrap;
    border-top: 1px solid #eee;
    border-bottom: 1px solid #eee;
  }
  
  .review-modal-date {
    font-size: 14px;
    color: #999;
    text-align: right;
    margin-top: 20px;
  }
  
  @media (max-width: 768px) {
    .review-modal {
      width: 95%;
      padding: 30px;
      margin: 20px;
    }
  
    .review-modal-author {
      font-size: 24px;
      margin-bottom: 10px;
    }
  
    .review-modal-rating .star {
      font-size: 22px;
    }
  
    .modal-close-button {
      top: 8px !important;
      right: 8px !important;
      padding: 0 !important;
      margin: 0 !important;
      border-radius: 0 !important;
      background: none !important;
      width: auto !important;
      height: auto !important;
    }
  }
  
  @media (max-width: 480px) {
    .review-modal {
      width: 100%;
      height: 100vh;
      margin: 0;
      border-radius: 0;
      padding: 20px;
    }
  
    .review-modal-author {
      font-size: 22px;
    }
  
    .review-modal-text {
      font-size: 15px;
      line-height: 1.5;
    }
  

  } 