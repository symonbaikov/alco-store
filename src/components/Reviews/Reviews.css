.reviews-section {
    width: 100%;
    height: 500px;
    padding: 4rem 0;
    background-color: #f8f9fa;
    color: black;
    margin-bottom: 60px;
    margin-top: 60px;
  }
  
  .container {
    max-width: 1500px;
    margin: 0 auto;
    padding: 0 1rem;
  }
  .chat-icon-btn {
    background: none;
    border: none;
    padding: 8px;
    cursor: pointer;
    color: #858383;
    transition: color 0.3s ease;
    display: flex;
    align-items: center;
    justify-content: center;
    min-width: 32px; /* Дайте достаточно пространства */
    min-height: 32px;
  }
  .reviews-title {
    font-weight: 500;
    color: #333;
    font-family: 'Poppins', sans-serif;
    font-size: 2rem;
  }
  .reviews-header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 40px;
    color: black;
    font-weight: 400;
  }

  .reviews-actions {
    display: flex;
    align-items: center;
    gap: 20px;
  }

  .leave-review-link {
    display: flex;
    align-items: center;
    gap: 8px;
    text-decoration: none;
    color: #858383;
    background-color: transparent;
    font-size: 14px;
    transition: color 0.3s ease;
  }

  .chat-icon-btn {
    color: #858383;
    transition: color 0.3s ease;
  }

  .chat-icon-btn svg {
    width: 34px;
    height: 34px;
  }

  .all-reviews-link {
    color: #858383;
    font-size: 13px;
    text-decoration: none;
    transition: color 0.3s ease;
  }

  .all-reviews-link:hover {
    color: #dc3545;
  }
  
  .leave-review-btn {
    padding: 10px 20px;
    background: transparent;
    cursor: pointer;
    transition: all 0.3s ease;
    color: black;
  }
  
  .reviews-slider {
    display: flex;
    align-items: center;
    justify-content: center;
    gap: 2rem;
    margin: 2rem 0;
    max-width: 1300px;
    margin: 0 auto;
  }
  .rating {
    margin-bottom: 1rem;
    display: flex;
    align-items: right;
    justify-content: right;
  }
  .review-card {
    padding: 2rem;
    width: 100%;
    max-width: 1500px;
    min-height: 300px;
    display: flex;
    flex-direction: column;
    align-items: left;
    justify-content: left;
    text-align: left;
  }
  

  
  .star {
    color: #ddd;
    font-size: 24px;
    margin: 0 2px;
  }
  
  .star.filled {
    color: #ffd700;
  }
  
  .review-author {
    font-size: 1.5rem;
    margin: 1rem 0;
    color: #333;
  }
  
  .review-text {
    font-size: 1.1rem;
    line-height: 1.6;
    color: #666;
    margin: 0;
    max-width: 1000px;
  }
  
  .slider-arrow {
    font-size: 30px;
    background: transparent;
    border: none;
    width: 55px;
    height: 50px;
    cursor: pointer;
    padding: 0px;
    background-color: white;
    border-radius: 5px;
    color: #333;
    transition: all 0.3s ease;
  }
  
  .slider-arrow:hover {
    color: white;
    background-color: #8b0000 ;
  }
  
  .read-more-btn {
    padding: 8px 20px;
    background: transparent;
    border: 1px solid #333;
    cursor: pointer;
    margin-top: 20px;
    transition: all 0.3s ease;
  }
  
  .read-more-btn:hover {
    background: #333;
    color: white;
  }
  
  @media (max-width: 768px) {
    .reviews-section {
      padding: 2rem 0;
      height: auto;
      margin-bottom: 20px;
      margin-top: 20px;
      min-height: 350px;
    }
  
    .container {
      padding: 0 1rem;
    }
  
    .reviews-header {
      flex-direction: column;
      gap: 10px;
      text-align: center;
      margin-bottom: 1.5rem;
    }
  
    .reviews-title {
      font-size: 1.5rem;
      margin-bottom: 0.5rem;
    }
  
    .reviews-actions {
      flex-direction: column;
      gap: 10px;
    }
  
    .reviews-slider {
      gap: 1rem;
      margin: 1rem 0;
    }
  
    .review-card {
      padding: 1.5rem;
      min-height: 200px;
    }
  
    .review-author {
      font-size: 1.2rem;
      margin: 0.5rem 0;
    }
  
    .review-text {
      font-size: 1rem;
      line-height: 1.4;
    }
  
    .rating {
      justify-content: center;
      margin-bottom: 0.5rem;
    }
  
    .star {
      font-size: 20px;
    }
  
    .slider-arrow {
      width: 40px;
      height: 40px;
      font-size: 24px;
    }
  }

  @media (max-width: 480px) {
    .reviews-section {
      padding: 1.5rem 0;
      min-height: 300px;
    }
  
    .container {
      padding: 0 0.75rem;
    }
  
    .reviews-title {
      font-size: 1.25rem;
    }
  
    .review-card {
      padding: 1rem;
      min-height: 180px;
    }
  
    .review-author {
      font-size: 1.1rem;
    }
  
    .review-text {
      font-size: 0.95rem;
      line-height: 1.3;
    }
  
    .star {
      font-size: 18px;
    }
  
    .slider-arrow {
      width: 35px;
      height: 35px;
      font-size: 20px;
    }
  
    .leave-review-link {
      font-size: 13px;
    }
  
    .all-reviews-link {
      font-size: 12px;
    }
  }

  @media (max-width: 375px) {
    .reviews-section {
      padding: 1rem 0;
      min-height: 280px;
    }
  
    .review-card {
      padding: 0.75rem;
      min-height: 160px;
    }
  
    .review-author {
      font-size: 1rem;
    }
  
    .review-text {
      font-size: 0.9rem;
      line-height: 1.2;
    }
  
    .star {
      font-size: 16px;
    }
  
    .slider-arrow {
      width: 30px;
      height: 30px;
      font-size: 18px;
    }
  }

  .loading, .error, .no-reviews {
    text-align: center;
    padding: 40px 0;
    font-size: 18px;
    color: #333;
  }
  
  .error {
    color: #dc3545;
  }
  
  .no-reviews {
    color: #666;
  }

  .error-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    min-height: 400px;
  }

  .broken-bottle-image {
    width: 500px;
    height: auto;
    margin-bottom: 2rem;
    opacity: 0.8;
    animation: shatter 0.5s ease-out;
  }

  .error-message {
    color: #dc3545;
    font-size: 1.2rem;
    margin: 1rem 0;
  }

  @keyframes shatter {
    0% {
      transform: scale(0.5) rotate(-10deg);
      opacity: 0;
    }
    50% {
      transform: scale(1.1) rotate(5deg);
    }
    100% {
      transform: scale(1) rotate(0);
      opacity: 0.8;
    }
  }

  /* Стили для модального окна с формой отзывов */
  .review-form {
    max-width: 500px;
    margin: 0 auto;
    padding: 30px;
    background: transparent;
    position: relative;
  }

  .modal-title {
    font-size: 24px;
    color: #333;
    text-align: left;
    margin-bottom: 25px;
    padding-right: 30px;
    font-weight: 400;
  }

  .form-group {
    margin-bottom: 20px;
  }

  .form-group label {
    display: block;
    margin-bottom: 8px;
    color: #666;
    font-weight: 400;
    font-size: 14px;
  }

  .form-group input,
  .form-group textarea {
    width: 100%;
    padding: 12px;
    border: 1px solid #e0e0e0;
    border-radius: 4px;
    font-size: 14px;
    color: #333;
    background: #fff;
    transition: all 0.3s ease;
  }

  .form-group input:focus,
  .form-group textarea:focus {
    outline: none;
    border-color: #8b0000;
    box-shadow: 0 0 0 1px rgba(139, 0, 0, 0.1);
  }

  .form-group input::placeholder,
  .form-group textarea::placeholder {
    color: #999;
    opacity: 1;
  }

  .form-group textarea {
    resize: vertical;
    min-height: 120px;
    background: #fff;
    color: #333;
  }

  .form-group input[type="file"] {
    display: none;
  }

  .file-upload-label {
    display: inline-flex;
    align-items: center;
    gap: 8px;
    padding: 10px 15px;
    background: #fff;
    border: 1px solid #e0e0e0;
    border-radius: 4px;
    cursor: pointer;
    color: #666;
    font-size: 14px;
    transition: all 0.3s ease;
  }

  .file-upload-label:hover {
    background: #f5f5f5;
  }

  .file-upload-label .attach-icon {
    font-size: 20px;
    color: #666;
  }

  .file-name-display {
    display: block;
    margin-top: 5px;
    font-size: 12px;
    color: #999;
  }

  .form-actions {
    display: flex;
    gap: 15px;
    margin-top: 25px;
    justify-content: flex-start;
  }

  .submit-btn,
  .cancel-btn {
    border: none;
    font-size: 20px;
    cursor: pointer;
    transition: all 0.3s ease;
    text-transform: none;
  }

  .submit-btn {
    background-color: #8b0000;
    color: white;
  }

  .submit-btn:hover {
    background-color: #660000;
  }

  .cancel-btn {
    background-color: #fff;
    color: #666;
    border: 1px solid #e0e0e0;
  }

  .cancel-btn:hover {
    background-color: #f5f5f5;
  }

  .modal-close-button {
    position: absolute;
    top: 10px;
    right: 10px;
    background: none;
    border: none;
    cursor: pointer;
    padding: 5px;
    display: flex;
    align-items: center;
    justify-content: center;
    z-index: 1000;
  }

  .modal-close-button svg {
    width: 20px;
    height: 20px;
    color: #999;
    transition: color 0.3s ease;
  }

  .modal-close-button:hover svg {
    color: #666;
  }

  .star-button {
    background: none;
    border: none;
    padding: 0;
    cursor: pointer;
    transition: transform 0.2s ease;
  }

  .star {
    width: 32px;
    height: 32px;
    color: #ddd;
  }

  .star.filled {
    color: #ffd700;
  }

  .error-message {
    color: #dc3545;
    font-size: 12px;
    margin-top: 5px;
  }

  /* Стили для звезд рейтинга */
  .rating-group {
    margin: 15px 0;
  }

  .stars-container {
    display: flex;
    gap: 8px;
    margin-top: 10px;
  }

  .star-button {
    background: none;
    border: none;
    padding: 0;
    cursor: pointer;
    color: white;
    font-size: 24px;
    transition: transform 0.2s ease;
  }

  .star-button:hover {
    transform: scale(1.1);
  }

  .star {
    color: rgba(255, 255, 255, 0.5);
    font-size: 24px;
  }

  .star.filled {
    color: #ffd700;
  }

  .error-message {
    color: #ff6b6b;
    font-size: 12px;
    margin-top: 5px;
  }

  /* Стили для чекбокса согласия */
  .consent-group {
    margin: 25px 0;
  }

  .consent-label {
    display: flex;
    align-items: flex-start;
    gap: 10px;
    cursor: pointer;
    font-size: 14px;
    color: #666;
    line-height: 1.4;
  }

  .consent-label input[type="checkbox"] {
    margin-top: 3px;
  }

  /* Стили для метки "необязательно" */
  .optional-label {
    color: #999;
    font-size: 12px;
    margin-left: 8px;
  }

  .review-modal-title {
    color: #000 !important;
  }

  @media (max-width: 430px) {
    .review-form {
      max-height: 100vh !important;
      overflow-y: auto !important;
    }
  }
