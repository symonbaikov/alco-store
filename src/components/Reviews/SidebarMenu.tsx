import React from 'react';
import { useLocation } from 'react-router-dom';

const menuItems = [
  { label: 'О компании', href: '/about' },
  { label: 'Сотрудничество', href: '/cooperation' },
  { label: 'Аренда', href: '/rent' },
  { label: 'Отзывы', href: '/reviews' },
  { label: 'Лицензии', href: '/licenses' },
  { label: 'Документы', href: '/documents' },
  { label: 'Реквизиты', href: '/requisites' },
  { label: 'Контакты', href: '/contacts' },
];

export default function SidebarMenu() {
  const location = useLocation();
  return (
    <nav className="sidebar-menu">
      <div className="menu-title">Меню</div>
      {menuItems.map(item => (
        <a
          key={item.href}
          href={item.href}
          className={
            'menu-item' + (location.pathname.startsWith(item.href) ? ' active' : '')
          }
        >
          {item.label}
        </a>
      ))}
    </nav>
  );
} 