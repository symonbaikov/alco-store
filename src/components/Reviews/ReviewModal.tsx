import React from 'react';
import Modal from '../Modal/Modal';
import { useTranslation } from 'react-i18next';
import StarIcon from '@mui/icons-material/Star';
import StarBorderIcon from '@mui/icons-material/StarBorder';
import './ReviewModal.css';

interface ReviewModalProps {
  isOpen: boolean;
  onClose: () => void;
  review: {
    author: string;
    text: string;
    rating: number;
    createdAt: string;
  } | null;
}

export const ReviewModal: React.FC<ReviewModalProps> = ({ isOpen, onClose, review }) => {
  const { t } = useTranslation();

  if (!review) return null;

  const renderStars = (rating: number) => {
    return [...Array(5)].map((_, index) => (
      <span key={index}>
        {index < rating ? (
          <StarIcon className="star filled" />
        ) : (
          <StarBorderIcon className="star" />
        )}
      </span>
    ));
  };

  return (
    <Modal isOpen={isOpen} onClose={onClose}>
      <div className="review-modal">
        <div className="review-modal-header">
          <h2 className="review-modal-author">{review.author}</h2>
          <div className="review-modal-rating">
            {renderStars(review.rating)}
          </div>
          <button className="modal-close-button" onClick={onClose} aria-label={t('common.close')}>
            <svg width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2.5" strokeLinecap="square" strokeLinejoin="miter">
              <line x1="5" y1="5" x2="19" y2="19" />
              <line x1="19" y1="5" x2="5" y2="19" />
            </svg>
          </button>
        </div>
        <div className="review-modal-content">
          <div className="review-modal-text">
            {review.text}
          </div>
          <div className="review-modal-date">
            {new Date(review.createdAt).toLocaleDateString()}
          </div>
        </div>
      </div>
    </Modal>
  );
}; 