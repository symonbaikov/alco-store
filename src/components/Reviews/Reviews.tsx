// /Users/symonbaikov/Projects/alco-store/src/components/Reviews/Reviews.tsx
import React, { useState } from "react";
import { useTranslation } from "react-i18next";
import { useReviews } from "../../hooks/useReviews";
import ChatIcon from "@mui/icons-material/ChatOutlined";
import { Loader } from "lucide-react";
import Modal from "../Modal/Modal.tsx";
import { ReviewForm } from "./ReviewForm";
import toast from 'react-hot-toast';
import { useAuthContext } from "../../context/AuthContext";
import "./Reviews.css"; // Make sure to add styles for the new elements here
import { Link } from "react-router-dom";

export const Reviews: React.FC = () => {
  const { t } = useTranslation();
  const [currentReview, setCurrentReview] = useState(0);
  const { reviews, loading, error, fetchReviews } = useReviews();
  const [isReviewModalOpen, setIsReviewModalOpen] = useState(false);
  const { isLoggedIn, user } = useAuthContext();

  const openReviewModal = () => {
    if (!isLoggedIn) {
      // Сохраняем состояние модального окна в localStorage
      localStorage.setItem('pendingReview', 'true');
      // Открываем модальное окно авторизации
      window.dispatchEvent(new CustomEvent('openAuth'));
      return;
    }
    setIsReviewModalOpen(true);
  };

  const closeReviewModal = () => setIsReviewModalOpen(false);

  // Проверяем наличие сохраненного отзыва при монтировании компонента
  React.useEffect(() => {
    const pendingReview = localStorage.getItem('pendingReview');
    if (pendingReview === 'true' && isLoggedIn) {
      setIsReviewModalOpen(true);
      localStorage.removeItem('pendingReview');
    }
  }, [isLoggedIn]);

  // Добавляем эффект для отслеживания данных пользователя
  React.useEffect(() => {
    if (isReviewModalOpen) {
      console.log('Opening review modal with user data:', user);
    }
  }, [isReviewModalOpen, user]);

  const handleReviewSubmit = async (formData: {
    name: string;
    email: string;
    message: string;
    file: File | null;
    rating: number;
  }) => {
    try {
      const loadingToast = toast.loading(t('reviews.submitting'));

      // Создаем FormData для отправки файла
      const form = new FormData();
      form.append('name', formData.name.trim());
      form.append('email', formData.email.trim());
      form.append('message', formData.message.trim());
      form.append('rating', formData.rating.toString());
      if (formData.file) {
        form.append('file', formData.file);
      }

      const response = await fetch('http://localhost:3001/api/reviews', {
        method: 'POST',
        credentials: 'include',
        body: form // Отправляем FormData вместо JSON
      });

      if (!response.ok) {
        const contentType = response.headers.get("content-type");
        if (contentType && contentType.includes("application/json")) {
          const errorData = await response.json();
          throw new Error(errorData.error || `HTTP error: ${response.status}`);
        }
        throw new Error(`HTTP error: ${response.status}`);
      }

      console.log('Ответ сервера:', response);
      
      closeReviewModal();
      
      toast.success(t('reviews.submitSuccess'), {
        id: loadingToast,
        duration: 3000
      });
      
      await fetchReviews();
    } catch (error) {
      console.error("Ошибка при отправке отзыва:", error);
      toast.error(t('reviews.submitError'), {
        duration: 4000
      });
    }
  };

  const renderStars = (rating: number) => {
    return [...Array(5)].map((_, index) => (
      <span key={index} className={`star ${index < rating ? "filled" : ""}`}>
        ★
      </span>
    ));
  };

  // --- Loading, Error, No Reviews states (remain the same) ---
  if (loading) {
    return (
      <section className="reviews-section">
        <div className="container">
          <div className="loading">
            <Loader className="animate-spin" />
          </div>
        </div>
      </section>
    );
  }

  if (error) {
    return (
      <section className="reviews-section">
        <div className="container error-container">
          <div className="error">
            <img
              src="/images/broken-bottle.png"
              alt={t("common.error")}
              className="broken-bottle-image"
            />
            <p className="error-message">
              {t("common.error")}: {error}
            </p>
          </div>
        </div>
      </section>
    );
  }

  if (!reviews.length) {
    return (
      <section className="reviews-section">
        <div className="container">
          <div className="no-reviews">{t("reviews.noReviews")}</div>
          {/* Optionally, still show the "Leave Review" button even if none exist yet */}
          <div className="reviews-actions only-leave-review">
            {" "}
            {/* Add specific class if needed */}
            <button
              type="button"
              className="leave-review-link"
              onClick={openReviewModal}
            >
              <ChatIcon className="chat-icon-btn" />
              {t("reviews.leaveFirstReview", "Be the first to leave a review")}
            </button>
          </div>
        </div>
      </section>
    );
  }
  // ---

  return (
    <>
      <section id="reviews" className="reviews-section">
        <div className="container">
          <div className="reviews-header">
            <h2 className="reviews-title">{t("reviews.title")}</h2>
            <div className="reviews-actions">
              <button
                type="button"
                className="leave-review-link"
                onClick={openReviewModal}
              >
                <ChatIcon className="chat-icon-btn" />
                {t("reviews.leaveReview")}
              </button>
              <Link to="/reviews" className="all-reviews-link">
                {t("reviews.allReviews")}
              </Link>
            </div>
          </div>

          <div className="reviews-slider">
            <button
              className="slider-arrow prev"
              onClick={() =>
                setCurrentReview((prev) =>
                  prev > 0 ? prev - 1 : reviews.length - 1
                )
              }
              aria-label={t("reviews.previous")}
            >
              ‹
            </button>

            <div className="review-card" role="article">
              <div
                className="rating"
                aria-label={`${t("reviews.rating")}: ${
                  reviews[currentReview].rating
                }/5`}
              >
                {renderStars(reviews[currentReview].rating)}
              </div>
              <h3 className="review-author">{reviews[currentReview].author}</h3>
              <p className="review-text">{reviews[currentReview].text}</p>
            </div>

            <button
              className="slider-arrow next"
              onClick={() =>
                setCurrentReview((prev) => (prev + 1) % reviews.length)
              }
              aria-label={t("reviews.next")}
            >
              ›
            </button>
          </div>
        </div>
      </section>

      <Modal isOpen={isReviewModalOpen} onClose={closeReviewModal}>
        <ReviewForm 
          onSubmit={handleReviewSubmit} 
          onClose={closeReviewModal}
          userData={user}
        />
      </Modal>
    </>
  );
};
