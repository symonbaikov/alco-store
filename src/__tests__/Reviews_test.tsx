import { render } from '@testing-library/react';
import { Reviews } from '../components/Reviews/Reviews';

// Мокаем useReviews
jest.mock('../hooks/useReviews', () => ({
  useReviews: () => ({
    reviews: [
      {
        id: 1,
        author: 'Иван Иванов',
        text: 'Отличный магазин!',
        rating: 5,
        createdAt: '2024-05-01T12:00:00Z',
      },
      {
        id: 2,
        author: 'Петр Петров',
        text: 'Быстрая доставка!',
        rating: 4,
        createdAt: '2024-05-02T12:00:00Z',
      },
    ],
    loading: false,
    error: null,
    fetchReviews: jest.fn(),
  })
}));

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key })
}));

jest.mock('../context/AuthContext', () => ({
  useAuthContext: () => ({ isLoggedIn: true, user: { firstName: 'Test', lastName: 'User', email: 'test@example.com' } })
}));

// Снапшот с отзывами
it('Reviews snapshot: renders with reviews', () => {
  const { asFragment } = render(<Reviews />);
  expect(asFragment()).toMatchSnapshot();
});

// Снапшот без отзывов
it('Reviews snapshot: renders with no reviews', () => {
  jest.mock('../hooks/useReviews', () => ({
    useReviews: () => ({
      reviews: [],
      loading: false,
      error: null,
      fetchReviews: jest.fn(),
    })
  }));
  const { asFragment } = render(<Reviews />);
  expect(asFragment()).toMatchSnapshot();
});

// Снапшот при загрузке
it('Reviews snapshot: renders loading state', () => {
  jest.mock('../hooks/useReviews', () => ({
    useReviews: () => ({
      reviews: [],
      loading: true,
      error: null,
      fetchReviews: jest.fn(),
    })
  }));
  const { asFragment } = render(<Reviews />);
  expect(asFragment()).toMatchSnapshot();
});

// Снапшот при ошибке
it('Reviews snapshot: renders error state', () => {
  jest.mock('../hooks/useReviews', () => ({
    useReviews: () => ({
      reviews: [],
      loading: false,
      error: 'Ошибка сервера',
      fetchReviews: jest.fn(),
    })
  }));
  const { asFragment } = render(<Reviews />);
  expect(asFragment()).toMatchSnapshot();
}); 