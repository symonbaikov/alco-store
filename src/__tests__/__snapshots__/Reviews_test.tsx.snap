// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews snapshot: renders error state 1`] = `
<DocumentFragment>
  <section
    class="reviews-section"
    id="reviews"
  >
    <div
      class="container"
    >
      <div
        class="reviews-header"
      >
        <h2
          class="reviews-title"
        >
          reviews.title
        </h2>
        <div
          class="reviews-actions"
        >
          <button
            class="leave-review-link"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium chat-icon-btn css-1umw9bq-MuiSvgIcon-root"
              data-testid="ChatOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 4h16v12H5.17L4 17.17zm0-2c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm2 10h8v2H6zm0-3h12v2H6zm0-3h12v2H6z"
              />
            </svg>
            reviews.leaveReview
          </button>
          <a
            class="all-reviews-link"
            href="/reviews"
          >
            reviews.allReviews
          </a>
        </div>
      </div>
      <div
        class="reviews-slider"
      >
        <button
          aria-label="reviews.previous"
          class="slider-arrow prev"
        >
          ‹
        </button>
        <div
          class="review-card"
          role="article"
        >
          <div
            aria-label="reviews.rating: 5/5"
            class="rating"
          >
            <span
              class="star filled"
            >
              ★
            </span>
            <span
              class="star filled"
            >
              ★
            </span>
            <span
              class="star filled"
            >
              ★
            </span>
            <span
              class="star filled"
            >
              ★
            </span>
            <span
              class="star filled"
            >
              ★
            </span>
          </div>
          <h3
            class="review-author"
          >
            Иван Иванов
          </h3>
          <p
            class="review-text"
          >
            Отличный магазин!
          </p>
        </div>
        <button
          aria-label="reviews.next"
          class="slider-arrow next"
        >
          ›
        </button>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Reviews snapshot: renders loading state 1`] = `
<DocumentFragment>
  <section
    class="reviews-section"
    id="reviews"
  >
    <div
      class="container"
    >
      <div
        class="reviews-header"
      >
        <h2
          class="reviews-title"
        >
          reviews.title
        </h2>
        <div
          class="reviews-actions"
        >
          <button
            class="leave-review-link"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium chat-icon-btn css-1umw9bq-MuiSvgIcon-root"
              data-testid="ChatOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 4h16v12H5.17L4 17.17zm0-2c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm2 10h8v2H6zm0-3h12v2H6zm0-3h12v2H6z"
              />
            </svg>
            reviews.leaveReview
          </button>
          <a
            class="all-reviews-link"
            href="/reviews"
          >
            reviews.allReviews
          </a>
        </div>
      </div>
      <div
        class="reviews-slider"
      >
        <button
          aria-label="reviews.previous"
          class="slider-arrow prev"
        >
          ‹
        </button>
        <div
          class="review-card"
          role="article"
        >
          <div
            aria-label="reviews.rating: 5/5"
            class="rating"
          >
            <span
              class="star filled"
            >
              ★
            </span>
            <span
              class="star filled"
            >
              ★
            </span>
            <span
              class="star filled"
            >
              ★
            </span>
            <span
              class="star filled"
            >
              ★
            </span>
            <span
              class="star filled"
            >
              ★
            </span>
          </div>
          <h3
            class="review-author"
          >
            Иван Иванов
          </h3>
          <p
            class="review-text"
          >
            Отличный магазин!
          </p>
        </div>
        <button
          aria-label="reviews.next"
          class="slider-arrow next"
        >
          ›
        </button>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Reviews snapshot: renders with no reviews 1`] = `
<DocumentFragment>
  <section
    class="reviews-section"
    id="reviews"
  >
    <div
      class="container"
    >
      <div
        class="reviews-header"
      >
        <h2
          class="reviews-title"
        >
          reviews.title
        </h2>
        <div
          class="reviews-actions"
        >
          <button
            class="leave-review-link"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium chat-icon-btn css-1umw9bq-MuiSvgIcon-root"
              data-testid="ChatOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 4h16v12H5.17L4 17.17zm0-2c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm2 10h8v2H6zm0-3h12v2H6zm0-3h12v2H6z"
              />
            </svg>
            reviews.leaveReview
          </button>
          <a
            class="all-reviews-link"
            href="/reviews"
          >
            reviews.allReviews
          </a>
        </div>
      </div>
      <div
        class="reviews-slider"
      >
        <button
          aria-label="reviews.previous"
          class="slider-arrow prev"
        >
          ‹
        </button>
        <div
          class="review-card"
          role="article"
        >
          <div
            aria-label="reviews.rating: 5/5"
            class="rating"
          >
            <span
              class="star filled"
            >
              ★
            </span>
            <span
              class="star filled"
            >
              ★
            </span>
            <span
              class="star filled"
            >
              ★
            </span>
            <span
              class="star filled"
            >
              ★
            </span>
            <span
              class="star filled"
            >
              ★
            </span>
          </div>
          <h3
            class="review-author"
          >
            Иван Иванов
          </h3>
          <p
            class="review-text"
          >
            Отличный магазин!
          </p>
        </div>
        <button
          aria-label="reviews.next"
          class="slider-arrow next"
        >
          ›
        </button>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Reviews snapshot: renders with reviews 1`] = `
<DocumentFragment>
  <section
    class="reviews-section"
    id="reviews"
  >
    <div
      class="container"
    >
      <div
        class="reviews-header"
      >
        <h2
          class="reviews-title"
        >
          reviews.title
        </h2>
        <div
          class="reviews-actions"
        >
          <button
            class="leave-review-link"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium chat-icon-btn css-1umw9bq-MuiSvgIcon-root"
              data-testid="ChatOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 4h16v12H5.17L4 17.17zm0-2c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm2 10h8v2H6zm0-3h12v2H6zm0-3h12v2H6z"
              />
            </svg>
            reviews.leaveReview
          </button>
          <a
            class="all-reviews-link"
            href="/reviews"
          >
            reviews.allReviews
          </a>
        </div>
      </div>
      <div
        class="reviews-slider"
      >
        <button
          aria-label="reviews.previous"
          class="slider-arrow prev"
        >
          ‹
        </button>
        <div
          class="review-card"
          role="article"
        >
          <div
            aria-label="reviews.rating: 5/5"
            class="rating"
          >
            <span
              class="star filled"
            >
              ★
            </span>
            <span
              class="star filled"
            >
              ★
            </span>
            <span
              class="star filled"
            >
              ★
            </span>
            <span
              class="star filled"
            >
              ★
            </span>
            <span
              class="star filled"
            >
              ★
            </span>
          </div>
          <h3
            class="review-author"
          >
            Иван Иванов
          </h3>
          <p
            class="review-text"
          >
            Отличный магазин!
          </p>
        </div>
        <button
          aria-label="reviews.next"
          class="slider-arrow next"
        >
          ›
        </button>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
